### PR TITLE
fixed delta peak timestamp problem

### PR DESCRIPTION
**Pull request by FaroutYLq**
**What is the problem / what does the code in this PR do**
The (lone hits) delta peaks added to `merged_s2s` waveforms have wrong time stamp. For example, in the example `merged_s2s` waveforms below, the red shadow marks peaklets while the blue shadow marks where we expect the lone hits, given the time stamps directly from `lone_hits` datatype. You can see the lone hits show up in the wrong time.
image
image
image

This PR did a minimal fix to this issue.

**Can you briefly describe how it works?**
```
index = (p['time'] - lh_i['time'])//p['dt']
```
is the problem. Apparently, lone hits' time should show up only after `merged_s2s` start. This implementation means that `index` is always negative, leading to a reversed ordering thus mirrored relative timestamp.

**Can you give a minimal working example (or illustrate with a figure)?**
I think this fix is simple enough to skip this part...

### CODE REVIEW

**Comment by JoranAngevaare** [comment]
Great catch, must have been introduced in https://github.com/AxFoundation/strax/pull/506!

Thanks a lot Lanqing!

One day we may try again to not merge lone hits in te first place, especially if they are after pulses.

**Review by JoranAngevaare — APPROVED** [review]

**Comment by WenzDaniel** [comment]
Thanks for catching Lanqing. 

Rearding 

> One day we may try again to not merge lone hits in te first place, especially if they are after pulses.

One has to be a bit careful to not introduce in that case a z- and area dependent bias. This was one of the issues in the 1T Ar37 analysis as far as I remember.

**Comment by WenzDaniel** [comment]
Can you update the test such that we do not run into this issue ever again https://github.com/AxFoundation/strax/blob/d6b3575df08f639cd4d054f116bb57f3cd2db92e/tests/test_peak_merging.py#L56 ?

File: tests/test_peak_merging.py [hunk 1]
```diff
@@ -51,12 +51,18 @@ def test_replace_merged(intervals, merge_instructions):
...
+                  )
 @hypothesis.settings(deadline=None)
-def test_add_lone_hits(hits, peak_length):
+def test_add_lone_hits(hits, time_offset, peak_length, dt):
     peak = np.zeros(1, dtype=strax.peak_dtype())
+    peak['time'] = time_offset
+    hits['time'] += time_offset
     peak['length'] = peak_length
-    peak['dt'] = 1
+    hits['area'] = 1
```
Comment (JoranAngevaare) [inline tests/test_peak_merging.py:64]: @WenzDaniel if we had realized this - the test would have worked. Now it always added zeros.

**Pull request merged**